modules/improvedigitalBidAdapter.js

```diff
@@ -216,8 +228,37 @@ export const spec = {
             syncs.push(syncElement);
           }
         });
+        // FOR PBS MODE
+        const seatBids = deepAccess(response, 'body.seatbid', []);
```

Comment:
This logic is wrong for several reasons:
1) It only syncs when there's a bid from Improve. We should also sync when there's no bid.
2) It relies on a bid impid - syncs are separated and independent from any particular request
3) This is iframe sync and runs from the "if pixelEnabled"
4) It doesn't check that iframe syncs are actually enabled

The logic should be like this: if pbs mode == true and iframe enabled => run iframe sync. else if pixelenabled, run the non-PBS logic.
You can store pbs mode value in buildRequests in a syncStore like bluebilly has it. It is enough if one ad unit has pbs enable to set pbs=true in the syncStore.